test-suite/shared/src/test/scala/org/scalajs/testsuite/compiler/LongTest.scala

```diff
@@ -0,0 +1,140 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.compiler
+
+import org.junit.Test
+import org.junit.Assert._
+
+/** Tests the compiler re-patching of native longs to
+ *  scala.scalajs.runtime.Long
+ *  see org.scalajs.testsuite.jsinterop.RuntimeLongTest
+ *  for a test of the implementation itself
+ */
+class LongTest {
+  @Test def `should_correctly_handle_literals`(): Unit = {
+    assertTrue(5L + 100L == 105L)
```

Comment:
With JUnit, we can finally use `assertEquals` even for `Long` comparison. We should do that.